src/main/java/com/metamx/common/CompressionUtils.java

```diff
@@ -0,0 +1,481 @@
+            byteSource,
+            tmpFile,
+            shouldRetry,
+            DEFAULT_RETRY_COUNT
+        );
+        return unzip(tmpFile, outDir);
+      }
+      finally {
+        if (!tmpFile.delete()) {
+          log.warn("Could not delete zip cache at [%s]", tmpFile.toString());
+        }
+      }
+    }
+  }
+
+  /**
+   * Unzip the byteSource to the output directory. If cacheLocally is true, the byteSource is cached to local disk before unzipping.
+   * This may cause more predictable behavior than trying to unzip a large file directly off a network stream, for example.
+   *
+   * @param byteSource   The ByteSource which supplies the zip data
+   * @param outDir       The output directory to put the contents of the zip
+   * @param cacheLocally A boolean flag to indicate if the data should be cached locally
+   *
+   * @return A FileCopyResult containing the result of writing the zip entries to disk
+   *
+   * @throws IOException
+   */
+  public static FileUtils.FileCopyResult unzip(
+      final ByteSource byteSource,
+      final File outDir,
+      boolean cacheLocally
+  ) throws IOException
+  {
+    return unzip(byteSource, outDir, FileUtils.IS_EXCEPTION, cacheLocally);
+  }
+
+  /**
+   * Unzip the pulled file to an output directory. This is only expected to work on zips with lone files, and is not intended for zips with directory structures.
+   *
+   * @param pulledFile The file to unzip
+   * @param outDir     The directory to store the contents of the file.
+   *
+   * @return a FileCopyResult of the files which were written to disk
+   *
+   * @throws IOException
+   */
+  public static FileUtils.FileCopyResult unzip(final File pulledFile, final File outDir) throws IOException
+  {
+    if (!(outDir.exists() && outDir.isDirectory())) {
+      throw new ISE("outDir[%s] must exist and be a directory", outDir);
+    }
+    log.info("Unzipping file[%s] to [%s]", pulledFile, outDir);
+    final FileUtils.FileCopyResult result = new FileUtils.FileCopyResult();
+    try (final ZipFile zipFile = new ZipFile(pulledFile)) {
+      final Enumeration<? extends ZipEntry> enumeration = zipFile.entries();
+      while (enumeration.hasMoreElements()) {
+        final ZipEntry entry = enumeration.nextElement();
+        result.addFiles(
+            FileUtils.retryCopy(
+                new ByteSource()
+                {
+                  @Override
+                  public InputStream openStream() throws IOException
+                  {
+                    return new BufferedInputStream(zipFile.getInputStream(entry));
+                  }
+                },
+                new File(outDir, entry.getName()),
+                FileUtils.IS_EXCEPTION,
+                DEFAULT_RETRY_COUNT
+            ).getFiles()
+        );
+      }
+    }
+    return result;
+  }
+
+  // Use unzip(ByteStream, File) if possible
+  public static FileUtils.FileCopyResult unzip(InputStream in, File outDir) throws IOException
```

Comment:
I think it makes sense to document those as well, since they are new to java-util. Otherwise, we would just leave them in druid-api. We also made some changes to the return values.